### ISSUE TITLE
`ensureSemantics()` breaks `Tap*Details` emitted from `GestureDetector` on mouse click on web

### ISSUE THREAD

**Issue by JaffaKetchup** [flutter/flutter#188859]
### Steps to reproduce

Run the code sample below on the web (specifically I'm running in Chrome on Windows). Click around the screen with your mouse. Remove/add the `ensureSemantics()` invocation if required for comparison.

This is causing https://github.com/fleaflet/flutter_map/issues/2176. There are some similarities to https://github.com/flutter/flutter/issues/146063 which has been open for 2 years and linked to https://github.com/flutter-team-archive/engine/pull/47360 which has been closed.

### Expected results

The position should be accurate, as is without running `ensureSemantics()`.

### Actual results

The printed information does not change depending on the cursor position (the position is always the center of the screen). Note that the `kind` is "unknown" rather than mouse. 

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(const MyApp());
  SemanticsBinding.instance.ensureSemantics();
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: GestureDetector(
          onTapDown: print,
          child: const SizedBox.expand(
            child: ColoredBox(color: Colors.black),
          ),
        ),
      ),
    );
  }
}

```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[√] Flutter (Channel beta, 3.46.0-0.2.pre, on Microsoft Windows [Version 10.0.26220.8754], locale en-GB) [518ms]
    • Flutter version 3.46.0-0.2.pre on channel beta at C:\Users\lukas\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 0f5a2c4694 (5 days ago), 2026-06-25 22:15:21 -0700
    • Engine revision a24b1ea55d
    • Dart version 3.13.0 (build 3.13.0-167.1.beta)
    • DevTools version 2.59.0
    • Feature flags: enable-web, no-enable-linux-desktop, no-enable-macos-desktop, enable-windows-desktop, enable-android, enable-ios, cli-animations, enable-native-assets, enable-swift-package-manager, omit-legacy-version-file, enable-lldb-debugging,
      enable-uiscene-migration

[√] Windows Version (11 Home Insider Preview 64-bit, 25H2, 2009) [1,614ms]

[!] Android toolchain - develop for Android devices (Android SDK version 36.0.0-rc3) [3.4s]
    • Android SDK at C:\Users\lukas\AppData\Local\Android\sdk
    • Emulator version 35.2.10.0 (build_id 12414864) (CL:N/A)
    • Platform android-36, build-tools 36.0.0-rc3
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.5+-13047016-b750.29)
    ! Some Android licenses not accepted. To resolve this, run: flutter doctor --android-licenses

[√] Chrome - develop for the web [162ms]
    • CHROME_EXECUTABLE = C:\Program Files\Google\Chrome Dev\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.14.13 (August 2025)) [162ms]
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.14.36414.22
    • Windows 10 SDK version 10.0.26100.0

[√] Connected device (3 available) [341ms]
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.26220.8754]
    • Chrome (web)      • chrome  • web-javascript • unknown
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 150.0.4078.42

[√] Network resources [320ms]
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by mbcorona** [comment]
Hi @JaffaKetchup , thanks for reporting and providing a code sample.
This appears closely related to #146063 (and its duplicate #177314), which likely shares the same root cause in the web engine's semantics/pointer path

Routing to the web team to evaluate.
